src/features/courses/course-controller.ts

```diff
@@ -2454,9 +2456,10 @@ class CourseController {
                     await questions.asyncForEach(async (question) => {
                         if (_.isNil(question.grades)) throw new RederlyExtendedError('Impossible! Found an assessment question without a grade.');
                         const version = await courseRepository.getCurrentInstanceForQuestion({questionId: question.id, userId});
-                        question.webworkQuestionPath = version?.webworkQuestionPath ?? question.webworkQuestionPath;
-                        question.grades.randomSeed = version?.randomSeed ?? question.grades.randomSeed;
-                        question.problemNumber = version?.problemNumber ?? question.problemNumber;
+                        if (!_.isNil(version)){
+                            question.grades[0].getVersion(version);
```

Comment:
What's the guarantee that a question has grades? We should add length check or null chain this
```
question.grades[0]?.getVersion(version);
```
Also we are not awaiting the response
Lastly we are not using the response (unless it is updating the base model

Reply:
Grades are created on enrollment or on question add, no? We have this check in the data-integrity-checker

You are correct about the updates - I mistakenly thought it was an edit in place.

Reply:
reverted these because it was too cumbersome to ensure the preservation of includes